hoot-services/src/main/java/hoot/services/controllers/grail/GrailResource.java

```diff
@@ -665,22 +686,24 @@ public Response grailMetadata(@Context HttpServletRequest request,
      * a private OSM instance that has diverged from the public OSM
      * with private changes.
      *
-     * @param bbox The bounding box
+     * @param reqParams Contains info such as bbox, layerName, and if the
+     * user provided one, a custom query for pulling overpass data
      *
      * @return Job ID
      * used by the client for polling job status
      * Internally, this provides the map dataset name suffix
      */
-    @GET
+    @POST
     @Path("/pullrailsporttodb")
     @Produces(MediaType.APPLICATION_JSON)
     public Response pullRailsPortToDb(@Context HttpServletRequest request,
-            @QueryParam("bbox") String bbox,
-            @QueryParam("name") String layerName) {
+            GrailParams reqParams) {
 
         Users user = Users.fromRequest(request);
         advancedUserCheck(user);
 
+        String bbox = reqParams.getBounds();
+        String layerName = reqParams.getInput1();
```

Comment:
Is `Input1` the most descriptive name we can use?  Would `name` or `layerName' work better?

Reply:
We would still need a way to differentiate between 2 possible inputs

Reply:
This parameter is normally used to pass a mapid but we are using it here to specify a name we want to assign